hashicorp/dev-portal
# feat: mdx-headings component to replace global-ish CSS

## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Refactors nested generic selectors in `dev-dot-content` to be part of our heading MDX custom components. Moves heading MDX custom components out of `sidebar-sidecar-layout` and into `components`, since they're shared between docs and tutorials content.

## 🤷 Why

- To avoid nested generic selectors cascading unintentionally into custom MDX components
- To more clearly consolidate heading styles used in MDX content

## 🛠️ How

- Removes nested generic `& h{n}` and `.h{n}` styling from `dev-dot-content`
- Creates new `mdx-headings` component to replace the above styles in MDX content
- Removes related `TODO` to remove `dev-dot-content` from `product-downloads-view`
- Removes `dev-dot-content` `.h{n}` style import from `marketing-content` component in `product-root-docs-path-landing` 
    - incoming styles were largely overridden (namely `margin`), and only `h2` & `h3` were being used
    - equivalent styles have been backfilled in CSS

## 🧪 Testing

- Visit a docs MDX page, such as [/vault/docs/get-started/developer-qs][/vault/docs/get-started/developer-qs], and confirm headings look as expected (vs [upstream][UPSTREAM/vault/docs/get-started/developer-qs])
- Visit a tutorial MDX page, such as [/waypoint/tutorials/get-started-docker/get-started-ui][/waypoint/tutorials/get-started-docker/get-started-ui], and confirm headings look as expected (vs [upstream][UPSTREAM/waypoint/tutorials/get-started-docker/get-started-ui])
- Visit a product docs landing page, such as [/vault][/vault] or [/waypoint][/waypoint], and confirm headings look as expected (vs upstream [/vault][UPSTREAM/vault] or [/waypoint][UPSTREAM/waypoint])

## 💭 Anything else?

Not at the moment!

[task]: https://app.asana.com/0/0/1202072894754377/f
[preview]: https://dev-portal-git-zsmdx-headings-hashicorp.vercel.app/
[/vault]: https://dev-portal-git-zsmdx-headings-hashicorp.vercel.app/vault
[UPSTREAM/vault]: https://developer.hashicorp.com/vault
[/waypoint]: https://dev-portal-git-zsmdx-headings-hashicorp.vercel.app/waypoint
[UPSTREAM/waypoint]: https://developer.hashicorp.com/waypoint
[/vault/docs/get-started/developer-qs]: https://dev-portal-git-zsmdx-headings-hashicorp.vercel.app/vault/docs/get-started/developer-qs
[UPSTREAM/vault/docs/get-started/developer-qs]: https://developer.hashicorp.com/vault/docs/get-started/developer-qs
[/waypoint/tutorials/get-started-docker/get-started-ui]: https://dev-portal-git-zsmdx-headings-hashicorp.vercel.app/waypoint/tutorials/get-started-docker/get-started-ui
[UPSTREAM/waypoint/tutorials/get-started-docker/get-started-ui]: https://developer.hashicorp.com/waypoint/tutorials/get-started-docker/get-started-ui
